### PR TITLE
Update Cognito Provider

### PR DESCRIPTION
**Pull request by JamesSiebert**
Removed Cognito provider fields that can cause errors.
Errors were occurring when the Cognito user pool doesn't have these same fields defined.

### CODE REVIEW

File: src/Cognito/Provider.php [hunk 1]
```diff
@@ -86,13 +85,7 @@ protected function getUserByToken($token): array
      */
     protected function mapUserToObject(array $user)
     {
-        return (new User())->setRaw($user)->map([
-            'id'       => $user['sub'],
-            'nickname' => $user['nickname'],
-            'name'     => trim(Arr::get($user, 'given_name', '').' '.Arr::get($user, 'family_name', '')),
-            'email'    => $user['email'],
```
Comment (atymic) [inline src/Cognito/Provider.php:93]: Why not default these to null instead of removing them completely? this will be a BC break.

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/Cognito/Provider.php (hunk 1 above)
Comment (JamesSiebert) [reply src/Cognito/Provider.php:93]: Good call, I was thinking that having all of the fields in one user array may be cleaner but I think this is ok, nulls added. Thanks :)

File: src/Cognito/Provider.php [hunk 2]
```diff
@@ -87,11 +87,10 @@ protected function getUserByToken($token): array
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'       => $user['sub'],
-            'nickname' => $user['nickname'],
+            'id'       => $user['sub'] ?? null,
+            'nickname' => $user['nickname'] ?? null,
             'name'     => trim(Arr::get($user, 'given_name', '').' '.Arr::get($user, 'family_name', '')),
-            'email'    => $user['email'],
-            'avatar'   => null,
```
Comment (lucasmichot) [inline src/Cognito/Provider.php:94]: Can you re-add the avatar field ?

**Review by lucasmichot — CHANGES_REQUESTED** [review]

File: src/Cognito/Provider.php (hunk 2 above)
Comment (JamesSiebert) [reply src/Cognito/Provider.php:94]: Sorry forgot to add back in, done :)

**Review by lucasmichot — APPROVED** [review]

**Comment by JamesSiebert** [comment]
Hey :) is there anything else I need to do here?

**Pull request merged**